sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py

```diff
@@ -314,14 +288,16 @@ def transform_outbound_single_message(message, message_type):
     :rtype: EventData
     """
     try:
-        # EventData
         # pylint: disable=protected-access
-        return message._to_outgoing_message()  # type: ignore
+        # EventData
+        message._message = to_outgoing_amqp_message(message.raw_amqp_message)
```

Comment:
Yeah, so previously we had the EventData._to_outgoing_message(). In that method, we had the following:
```event_data.message = event_data.raw_amqp_message.to_outgoing_amqp_message()```.
Just to clarify what's happening, two consecutive changes were made here:
1) We no longer have the `to_outgoing_amqp_message` on the AmqpAnnotatedMessage, b/c we want to avoid adding pyamqp/uamqp specific code in AmqpAnnotatedMessage. So, we're constructing the outgoing amqp message here. At this point, this line would have looked like `message.message = to_outgoing_amqp_message(message.raw_amqp_message)`.
2) Next change made: Making the `EventData.message` ivar private and adding the `EventData.message` public property. Everywhere that we previously called `event_data.message`, we're now calling `event_data._message` instead. This is in preparation for adding LegacyMessage when we merge pyamqp. For right now, the EventData.message getter and setter are both = `self._message`. So technically, having this line be `message.message = ...` would be fine for now. However, b/c we likely won't be calling `message.message` when LegacyMessage is merged, I'm just mutating the internal message here.

Idk if I answered your original question and maybe I overexplained what you already knew, so please lmk if this doesn't make sense/you have concerns 😄